v1/daemon/write.go

```diff
@@ -15,19 +15,45 @@
 package daemon
 
 import (
-	"fmt"
+	"context"
+	"io"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
 
 	"github.com/google/go-containerregistry/name"
 	"github.com/google/go-containerregistry/v1"
+	"github.com/google/go-containerregistry/v1/tarball"
 )
 
+// API interface for testing.
+type ImageLoader interface {
+	ImageLoad(context.Context, io.Reader, bool) (types.ImageLoadResponse, error)
+}
+
+// This is a variable so we can override in tests.
+var getImageLoader = func() (ImageLoader, error) {
+	return client.NewEnvClient()
+}
+
 // WriteOptions are used to expose optional information to guide or
 // control the image write.
 type WriteOptions struct {
 	// TODO(dlorenc): What kinds of knobs does the daemon expose?
 }
 
 // Write saves the image into the daemon as the given reference.
-func Write(ref name.Reference, img v1.Image, wo WriteOptions) error {
-	return fmt.Errorf("NYI: daemon.Write(%v)", ref)
+func Write(ref name.Reference, img v1.Image, wo WriteOptions) (types.ImageLoadResponse, error) {
```

Comment:
This library's API shouldn't return some other library's types. Is there another way to express this?

Reply:
ah yeah that's a great point. I still want to expose the client's response to the user, so I'll change this to return `(string, error)`